packages/plugin-init/sources/index.ts

```diff
@@ -2,6 +2,16 @@ import {Plugin, SettingsType} from '@yarnpkg/core';
 
 import init                   from './commands/init';
 
+declare module '@yarnpkg/core' {
+  interface ConfigurationValueMap {
+    initLicense: string | null;
+    initScope: string | null;
+    initVersion: string | null;
+    initFields: Map<string, any>;
+    initEditorConfig: Map<string, any>;
+  }
```

Comment:
TIL! That's a really cool pattern - it would make a lot of sense to use it for hooks as well, this would guarantee that multiple plugins exposing the same hook would share the same prototype, since TS would error otherwise